integrations/satismeter/lib/index.js

```diff
@@ -66,10 +75,41 @@ SatisMeter.prototype.identify = function(identify) {
  * @param {Page} page
  */
 
-SatisMeter.prototype.page = function() {
-  window.satismeter({
-    writeKey: this.options.apiKey || this.options.token,
+SatisMeter.prototype.page = function(page) {
+  window.satismeter('page', {
+    userId: this.analytics.user().id(),
+    name: page.name(),
+    category: page.category(),
+    properties: page.properties()
+  });
+};
+
+/**
+ * Track.
+ *
+ * @api public
+ * @param {Track} track
+ */
+
+SatisMeter.prototype.track = function(track) {
+  window.satismeter('track', {
+    userId: this.analytics.user().id(),
```

Comment:
@E-Mud I think it's possible to track events also anonymously with Segment. Any ideas how can we support it as well?

Reply:
@jsedlacek In theory with [user's method for anonymousId](https://segment.com/docs/connections/sources/catalog/libraries/website/javascript/#retrieving-the-anonymous-id). But it says that if it doesn have an anonymousId that call will set it.

We could add something like:
```javascript
var userId = this.analytics.user().id();

if (!userId) {
  options.anonymousId = this.analytics.user().anonymousId();
}
```
But that will set the anonymousId if there wasn't one. I'm not sure if that could be a case, i.e. user not having either `userId` nor `anonymousId`. I think it's possible if there's no `identify` call. The call will set the anonymousId for the whole analytics "session", not just for our integration but I'm not happy with that.

I'll check what others do on these cases.

By the way, should we add anonymousId to `identify` aswell?

Reply:
Or maybe [Facade's anonymousId](https://facade.segment.com/#facadeanonymousid) is better option

Reply:
I see a mix of uses. I've been testing the differences and I think there's none. `track.anonymousId()` returns the same as `user.anonymousId()`. Even if I don't touch the user first to set the userId.

Reply:
@jsedlacek I just realized that if no userId and no anonymousId has been set (like if there no `identify` call) our widget will add an anonymousId. But before generating one it tries to get it first from analytics.js user:
```javascript
import cookie from 'cookie-cutter';
import uuid from 'uuid';

const COOKIE_NAME = 'sm_anonymous_id';

declare global {
  interface Window {
    analytics: any;
  }
}

function anonymousId() {
  if (window.analytics) {
    // Try to use anonymous id from Segment.com
    try {
      return window.analytics.user().anonymousId();
    } catch (ex) {}
  }

  let id: string = cookie.get(COOKIE_NAME);

  if (!id) {
    id = uuid();

    const now = new Date();
    const expires = new Date(now.getTime() + 365 * 24 * 3600 * 1000);

    cookie.set(COOKIE_NAME, id, { expires: expires, path: '/' });
  }

  return id;
}

export default anonymousId;
```
So for track/page/group calls it will attach that anonymousId automatically, without the need of specifying it in the integration.

It's a matter of whether we want to be more explicit in our integration code or not

Reply:
@E-Mud I think we can expect anonymous ID is always set. It's automatically generated by Segment. I'd suggest to always forward `anonymousId` to us.

Reply:
We can forward the anonymous ID even if the user ID is set. This is what Segment does itself when sending API calls to their own API endpoints.

Reply:
@E-Mud It's definitely better to pass it explicitly.

Reply:
@jsedlacek Done